### ISSUE TITLE
Unresolved reference ':'

### ISSUE THREAD

**Issue by Dierk** [IntelliJ-Frege/intellij-frege#67]
It appears that the colon cons operator is not recognized (?) in

```
main = do
    println $ 1 : []
```

even though the code compiles and runs fine

**Comment by kkarnauk** [comment]
Yes, I know about this and couldn't figure out how to fix it. The problem is that I couldn't find a place where the colon constructor is defined. Is that true that it's not defined in the usual way and hidden in compiler?

**Comment by Dierk** [comment]
the infix operator is declared in PreludeBase line 104 ...

**Comment by Dierk** [comment]
the colon will be rewritten as "Cons" in Mangle.fr line 76
`mangled ":"  = "Cons"`

**Comment by kkarnauk** [comment]
Okay, thanks! I'll take a look at that.

**Comment by Dierk** [comment]
BTW: if you run
`1:[]`
in the frege repl http://86.119.43.169:9999
and look at the java code with
`:java`
then you see
```
final public static PreludeBase.TList<Integer> frege_interpreter_var1 = PreludeBase.TList.DCons.<
  Integer
>mk(Thunk.<Integer>lazy(1), PreludeBase.TList.DList.<Integer>mk());
```
meaning the` : ` is translated to the `Cons` constructor of Frege's List type.

**Comment by Dierk** [comment]
Ha, I  just came across it again (as it is a quite often-used feature) and checked whether it has been reported before and it was - by me! lol.
